### PR TITLE
add hostname option

### PR DESCRIPTION
**Pull request by super-ienien**
This PR add the hostname options to options in addition to DISCOVERY_HOSTNAME env var. And update the doc.

### CODE REVIEW

**Comment by dashersw** [comment]
Thank you for the PR @super-ienien ! There's a little problem with the current version. Since `hostName` is a module-wide global variable and `Network` is a class, passing in two different `hostname` values will not work, in that it will override the value for the first `Network` instance as well. This was not a problem previously with the env var, because once set it can't change.

Could you update the PR and use `self.hostName = options.hostName || hostName` instead of `hostName` directly? You would also need to update `Network.prototype.send`.

**Comment by super-ienien** [comment]
Ok, it's done

File: lib/network.js [hunk 1]
```diff
@@ -32,6 +32,8 @@ function Network (options) {
     self.ignoreProcess  = (options.ignoreProcess ===  false) ? false : true;
     self.ignoreInstance = (options.ignoreInstance ===  false) ? false : true;
     self.redis          = options.redis;
+    self.hostName       = options.hostname  || hostname;
```
Comment (dashersw) [inline lib/network.js]: The original module scope variable is called `hostName`, not `hostname`. And there's an additional space before `||`.

I think this line should read `self.hostName       = options.hostname || hostName;`

**Comment by dashersw** [comment]
Thank you! Will release a new cote version with this.

**Comment by dashersw** [comment]
Released cote version 0.17.1.

**Pull request merged**